### ISSUE TITLE
After upgrading from 2.0.5 to 2.1.1 using the exists() operator on a date field in combination with $qb->addAnd() breaks

### ISSUE THREAD

**Issue by marcbln** [doctrine/mongodb-odm#2207]
### BC Break Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 2.1.1
| PHP Version | 7.4.5

#### Summary

After an upgrade from 2.0.5 to 2.1.1 using the exists() operator on a date field in combination with $qb->addAnd() breaks, please see example code below.

#### Previous behavior

On version 2.0.5 no exception was thrown and the QueryBuilder worked as expected, generating a working query.

#### Current behavior

An Exception is thrown when using `$expr->field('someDateField')->exists()` on a date field in combination with `$qb->addAnd($expr)`

#### How to reproduce

example code in a document repository:

```php
        $qb = $this->createQueryBuilder();
        $expr = $qb->expr();
        $expr->field('shippedTime')->exists(false);
        $qb->addAnd($expr);
```
with the shippedTime field in the document class defined as date:

```php
    /**
     * @MongoDB\Field(type="date")
     * @var \DateTime
     */
    private $shippedTime;
```

yields to an exception '_Could not convert "" to a date value_':

```json
{
    "type": "https://tools.ietf.org/html/rfc2616#section-10",
    "title": "An error occurred",
    "status": 500,
    "detail": "Could not convert \"\" to a date value",
    "class": "InvalidArgumentException",
    "trace": [
        {
            "namespace": "",
            "short_class": "",
            "class": "",
            "type": "",
            "function": "",
            "file": "/var/www/html/vendor/doctrine/mongodb-odm/lib/Doctrine/ODM/MongoDB/Types/DateType.php",
            "line": 65,
            "args": []
        },
        {
            "namespace": "Doctrine\\ODM\\MongoDB\\Types",
            "short_class": "DateType",
            "class": "Doctrine\\ODM\\MongoDB\\Types\\DateType",
            "type": "::",
            "function": "getDateTime",
            "file": "/var/www/html/vendor/doctrine/mongodb-odm/lib/Doctrine/ODM/MongoDB/Types/DateType.php",
            "line": 91,
            "args": [
                [
                    "boolean",
                    false
                ]
            ]
        },
        {
            "namespace": "Doctrine\\ODM\\MongoDB\\Types",
            "short_class": "DateType",
            "class": "Doctrine\\ODM\\MongoDB\\Types\\DateType",
            "type": "->",
            "function": "convertToDatabaseValue",
            "file": "/var/www/html/vendor/doctrine/mongodb-odm/lib/Doctrine/ODM/MongoDB/Persisters/DocumentPersister.php",
            "line": 1069,
            "args": [
                [
                    "boolean",
                    false
                ]
            ]
        },
        {
            "namespace": "Doctrine\\ODM\\MongoDB\\Persisters",
            "short_class": "DocumentPersister",
            "class": "Doctrine\\ODM\\MongoDB\\Persisters\\DocumentPersister",
            "type": "->",
            "function": "convertToDatabaseValue",
            "file": "/var/www/html/vendor/doctrine/mongodb-odm/lib/Doctrine/ODM/MongoDB/Persisters/DocumentPersister.php",
            "line": 1045,
            "args": [
                [
                    "string",
                    "shippedTime"
                ],
                [
                    "boolean",
                    false
                ]
            ]
        },
        {
            "namespace": "Doctrine\\ODM\\MongoDB\\Persisters",
            "short_class": "DocumentPersister",
            "class": "Doctrine\\ODM\\MongoDB\\Persisters\\DocumentPersister",
            "type": "->",
            "function": "convertToDatabaseValue",
            "file": "/var/www/html/vendor/doctrine/mongodb-odm/lib/Doctrine/ODM/MongoDB/Persisters/DocumentPersister.php",
            "line": 1015,
            "args": [
                [
                    "string",
                    "shippedTime"
                ],
                [
                    "array",
                    {
                        "$exists": [
                            "boolean",
                            false
                        ]
                    }
                ]
            ]
        },
        {
            "namespace": "Doctrine\\ODM\\MongoDB\\Persisters",
            "short_class": "DocumentPersister",
            "class": "Doctrine\\ODM\\MongoDB\\Persisters\\DocumentPersister",
            "type": "->",
            "function": "prepareQueryOrNewObj",
            "file": "/var/www/html/vendor/doctrine/mongodb-odm/lib/Doctrine/ODM/MongoDB/Query/Expr.php",
            "line": 603,
            "args": [
                [
                    "array",
                    {
                        "shippedTime": [
                            "array",
                            {
                                "$exists": [
                                    "boolean",
                                    false
                                ]
                            }
                        ]
                    }
                ]
            ]
        },
        {
            "namespace": "Doctrine\\ODM\\MongoDB\\Query",
            "short_class": "Expr",
            "class": "Doctrine\\ODM\\MongoDB\\Query\\Expr",
            "type": "->",
            "function": "getQuery",
            "file": "/var/www/html/vendor/doctrine/mongodb-odm/lib/Doctrine/ODM/MongoDB/Query/Expr.php",
            "line": 99,
            "args": []
        },
        {
            "namespace": "Doctrine\\ODM\\MongoDB\\Query",
            "short_class": "Expr",
            "class": "Doctrine\\ODM\\MongoDB\\Query\\Expr",
            "type": "::",
            "function": "Doctrine\\ODM\\MongoDB\\Query\\{closure}",
            "file": null,
            "line": null,
            "args": [
                [
                    "object",
                    "Doctrine\\ODM\\MongoDB\\Query\\Expr"
                ]
            ]
        },
        {
            "namespace": "",
            "short_class": "",
            "class": "",
            "type": "",
            "function": "array_map",
            "file": "/var/www/html/vendor/doctrine/mongodb-odm/lib/Doctrine/ODM/MongoDB/Query/Expr.php",
            "line": 101,
            "args": [
                [
                    "object",
                    "Closure"
                ],
                [
                    "array",
                    [
                        [
                            "object",
                            "Doctrine\\ODM\\MongoDB\\Query\\Expr"
                        ]
                    ]
                ]
            ]
        },
        {
            "namespace": "Doctrine\\ODM\\MongoDB\\Query",
            "short_class": "Expr",
            "class": "Doctrine\\ODM\\MongoDB\\Query\\Expr",
            "type": "->",
            "function": "addAnd",
            "file": "/var/www/html/vendor/doctrine/mongodb-odm/lib/Doctrine/ODM/MongoDB/Query/Builder.php",
            "line": 144,
            "args": [
                [
                    "object",
                    "Doctrine\\ODM\\MongoDB\\Query\\Expr"
                ]
            ]
        },
        {
            "namespace": "Doctrine\\ODM\\MongoDB\\Query",
            "short_class": "Builder",
            "class": "Doctrine\\ODM\\MongoDB\\Query\\Builder",
            "type": "->",
            "function": "addAnd",
            "file": "/var/www/html/src/App/Repository/Ebay/EbayOrderRepository.php",
            "line": 231,
            "args": [
                [
                    "object",
                    "Doctrine\\ODM\\MongoDB\\Query\\Expr"
                ]
            ]
        },
        "...."
    ]
}
```

**Comment by malarzm** [comment]
@marcbln can you please see if upgrading to 2.1.2 fixes your issue? It was released today and includes #2205 which fixed issue similar to yours

**Comment by marcbln** [comment]
An upgrade to version 2.1.2 fixed the issue. Thank you.
